modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/ProductConfigurationDTOConverter.java

```diff
@@ -39,7 +43,14 @@ public ProductConfiguration toDTO(DTOConverterContext dtoConverterContext)
 				fetchCPDefinitionInventoryByCPDefinitionId(
 					(Long)dtoConverterContext.getId());
 
-		if (cpDefinitionInventory == null) {
+		CPDAvailabilityEstimate cpdAvailabilityEstimate =
+			_cpdAvailabilityEstimateService.
+				fetchCPDAvailabilityEstimateByCPDefinitionId(
+					(Long)dtoConverterContext.getId());
+
+		if ((cpDefinitionInventory == null) &&
```

Comment:
Now that we don't check only inventory, we should do null checks for every setter that is depending on inventory